integration-tests/capacity/transactions.test.ts

```diff
@@ -111,38 +113,38 @@ describe("Capacity Transactions", function () {
                 });
             });
 
-            // A registered provider with Capacity but no tokens associated with the
-            // key should still be able to use polkadot UI to submit a capacity transaction.
-            // *All accounts will have at least an EXISTENTIAL_DEPOSIT = 1M.
-            // This test may still be valid if the txn cost is greater than 1M.
-            it("pays for a transaction with available capacity and no free tokens", async function () {
-                let delegatorKeys = createKeys("delegatorKeys");
-                let _delegatorMsaId = await ExtrinsicHelper.createMsa(delegatorKeys);
+            // *All keys should have at least an EXISTENTIAL_DEPOSIT = 1M.
+            it("fails to pay for transaction when key does has not met the min deposit", async function () {
+                let newProviderKeys = createKeys("newKeys");
 
-                let providerKeys = createKeys("providerKeys");
-                let providerId = await createMsaAndProvider(providerKeys, "UnstakeProvider");
+                await assert.doesNotReject(stakeToProvider(stakeKeys, stakeProviderId, 1n * dollars));
+
+                // Add new key
+                let newKeyPayload: AddKeyData = await generateAddKeyPayload({ msaId: new u64(ExtrinsicHelper.api.registry, stakeProviderId), newPublicKey: newProviderKeys.publicKey });
+                let addKeyData = ExtrinsicHelper.api.registry.createType("PalletMsaAddKeyData", newKeyPayload);
 
-                // Stake enough to cover the 2.8M grantDelegation cost
-                await assert.doesNotReject(stakeToProvider(stakeKeys, providerId, 3n * cents));
+                let ownerSig = signPayloadSr25519(stakeKeys, addKeyData);
+                let newSig = signPayloadSr25519(newProviderKeys, addKeyData);
+                const addPublicKeyOp = ExtrinsicHelper.addPublicKeyToMsa(stakeKeys, ownerSig, newSig, newKeyPayload);
+
+                const [publicKeyEvents] = await addPublicKeyOp.fundAndSend();
+                assert.notEqual(publicKeyEvents, undefined, 'should have added public key');
+
+                let delegatorKeys = createKeys("delegatorKeys");
+                let _delegatorMsaId = await ExtrinsicHelper.createMsa(delegatorKeys);
 
```

Comment:
Do we also need to verify createMsa() succeeded here?

Reply:
We don't, but it makes the code confusing. We don't because it's going to fail before reaching that code.